File: content-solutions/greedy/cookie-share.java
import java.util.*;

class Solution {
    public int solution(int[] greed, int[] cookies) {
        int[] g = greed.clone();
        int[] c = cookies.clone();
        Arrays.sort(g);
        Arrays.sort(c);
        int i = 0;
        for (int size : c) {
            if (i < g.length && size >= g[i]) i++;
        }
        return i;
    }
}

File: content-solutions/greedy/cookie-share.js
function solution(greed, cookies) {
  const g = [...greed].sort((a, b) => a - b);
  const c = [...cookies].sort((a, b) => a - b);
  let i = 0;
  for (const size of c) {
    if (i < g.length && size >= g[i]) i++;
  }
  return i;
}
